app/services/EventSyncService.ts

```diff
@@ -0,0 +1,68 @@
+import { dataRealmCommon } from './../database/dbquery/dataRealmCommon';
+import analytics from '@react-native-firebase/analytics';
+
+
+export const logEventToFirebase = async (analyticsInstance: any, event: any): Promise<void> => {
+  if(event.params!=undefined){
+  analyticsInstance.logEvent(event.name, event.params);
+  }else{
+    analyticsInstance.logEvent(event.name)
+  }
+}
+
+export const markEventAsSynchronized = async (realm: any, event: any): Promise<void> => {
+  if (!dataRealmCommon.isRealmClosed()) {
+    realm = await dataRealmCommon.openRealm();
+  }
+  realm.write(() => {
+    event.isSynchronized = true;
+  });
+
+}
+
+export const storeUnsyncedEvent = async (realm: any, eventData: any): Promise<any> => {
+  realm.write(() => {
+    const unsyncedEvent = realm.create('Event', {
+      ...eventData,
+      isSynchronized: false,
+    });
+    console.log('EventClick unsyncedEvent for category', unsyncedEvent);
+  });
+}
+
+export const logEvent = async (eventData: any, netInfo: boolean): Promise<any> => {
+  let realm: any;
+  if (!dataRealmCommon.isRealmClosed()) {
```

Comment:
This condition is a bit confusing. How should I read it? "If realm database is not closed then ..." we will open it. Why do we open if database is not closed?